.github/workflows/pull-request.yml

```diff
@@ -67,22 +67,72 @@ jobs:
           python .github/workflows/compare_translations.py --directory public/locales
 
 
-  Check-Changed-Files:
-    name: File count, sensitive files and branch check
+  Check-Unauthorized-Changes:
+    name: Checks if no unauthorized files are changed
     runs-on: ubuntu-latest
-    # needs: Code-Quality-Checks
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
-
-      - name: Set up Python
-        uses: actions/setup-python@v5
+      
+      - name: Get Changed Unauthorized files
+        id: changed-unauth-files
+        uses: tj-actions/changed-files@v40
         with:
-          python-version: 3.9
+          files: |
+            .github/**
+            env.example
+            .node-version
+            .husky/**
+            scripts/**
+            package.json
+            tsconfig.json
+            .gitignore
+            .eslintrc.json
+            .eslintignore
+            vite.config.ts
+            docker-compose.yaml
+            Dockerfile
+            CODEOWNERS
+            LICENSE
+            setup.ts
+            
+      - name: List all changed unauthorized files
+        if: steps.changed-unauth-files.outputs.any_changed == 'true' || steps.changed-unauth-files.outputs.any_deleted == 'true'
+        env: 
+          CHANGED_UNAUTH_FILES: ${{ steps.changed-unauth-files.outputs.all_changed_files }}
+        run: |
+          for file in ${CHANGED_UNAUTH_FILES}; do
+            echo "$file is unauthorized to change/delete"
+          done
+          exit 1
 
-      - name: Run Python script
+  Count-Changed-Files:
+    name: Checks if number of files changed is acceptable
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v40
+
+      - name: Echo number of changed files
+        env: 
+          CHANGED_FILES_COUNT: ${{ steps.changed-files.outputs.all_changed_files_count }}
         run: |
-          python .github/workflows/count_changed_files.py --base_branch "${{ github.base_ref }}" --pr_branch "${{ github.head_ref }}"
+          echo "Number of files changed: $CHANGED_FILES_COUNT"
+
+      - name: Check if the number of changed files is less than 20
+        if: steps.changed-files.outputs.all_changed_files_count > 20
```

Comment:
Set this to 100. We created it to alert us to merges between develop and main by newcomers. We mostly ignore it now because we sometimes have big PRs. They are usually under 100 files, so that would be a better number for the trigger.

Reply:
Got it. Thank you